Docs/troubleshooting.md

```diff
@@ -0,0 +1,236 @@
+
+| Variable   | Value | Explanation                                          |
+|------------|-------|------------------------------------------------------|
+| LOG_TO_HUB | true  | Log info are sent from the module to IoT Hub.        |
+|            | false | Log info is not sent to IoT Hub (default if omitted) |
+
+You can use VSCode, [IoTHub explorer](https://github.com/Azure/iothub-explorer) or [Device Explorer](https://github.com/Azure/azure-iot-sdk-csharp/tree/master/tools/DeviceExplorer) to monitor the log messages directly in IoT Hub if **LOG_TO_HUB** is set to **true**.
+
+Log in to the gateway and use `sudo docker logs LoRaWanNetworkSrvModule -f` to follow the logs if you are not logging to IoT Hub.
+
+| Variable       | Value | Explanation                             |
+|----------------|-------|-----------------------------------------|
+| LOG_TO_CONSOLE | true  | Log to docker logs (default if omitted) |
+|                | false | Does not log to docker logs             |
+
+## Customize the solution
+
+Have a look at the [LoRaEngine folder](/LoRaEngine) for more in details explanation.
+
+## Cloud to device message
+
+The solution support sending Cloud to device (C2D) messages to LoRa messages using [standard IoT Hub Sdks](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-messages-c2d). Cloud to device messages require a Fport message property being set or it will be refused (as shown in the figure below from the Azure Portal). 
+
+![C2D portal](/Docs/Pictures/cloudtodevice.png)
+
+The following tools can be used to send cloud to devices messages from Azure :
+
+* [Azure Portal](http://portal.azure.com) -> IoT Hub -> Devices -> message to device
+* [Device Explorer](https://github.com/Azure/azure-iot-sdk-csharp/tree/master/tools/DeviceExplorer)
+* [Visual Studio Code IoT Hub Extension](https://marketplace.visualstudio.com/items?itemName=vsciot-vscode.azure-iot-toolkit) 
+
+It is possible to add a 'Confirmed' message property set to true,in order to send the C2D message as ConfirmedDataDown to the LoRa device (as in picture above and below). You can enable additional message tracking options by setting the C2D message id to a value (C2D message ID is automatically populated with the Device Explorer tool used in the image below). 
+
+![C2D portal](/Docs/Pictures/sendC2DConfirmed.png)
+
+As soon as the device acknowledges the message, it will report it in the logs and as a message property named 'C2DMsgConfirmed' on a message upstream (or generate an empty message in case of an empty ack message). The value of the message property will be set to the C2D message id that triggered the response if not null, otherwise to 'C2D Msg Confirmation'. You can find here below a set of picture illustrating the response when the C2D message id was sent to the value '4d3d0cd3-603a-4e00-a441-74aa55f53401'.
```

Comment:
The first sentace is incomplete, or has a typo... speficially after the word "and".. it starts with as, but it ever gets finished... therefore is the word "as" a typo? perhaps you meant "adds", or "asks"? Not sure... 

Also in the brackets (we are still on the first sentance)... what is a "ack" message? acknowledgement? I am not sure.... 

The second sentance should start with the "If" for easiert reading... like this: 

If not null, the value of the message property will be set to the C2D message id that triggered the response, otherwise it will be set to 'C2D Msg Confirmation'.